components/motor/gpio/controlled.go

```diff
@@ -30,8 +30,6 @@ func (cm *controlledMotor) SetState(ctx context.Context, state []*control.Signal
 // State gets the state of the motor for the built-in control loop.
 func (cm *controlledMotor) State(ctx context.Context) ([]float64, error) {
 	ticks, _, err := cm.enc.Position(ctx, encoder.PositionTypeTicks, nil)
-	cm.mu.RLock()
-	defer cm.mu.RUnlock()
```

Comment:
to make your testing a bit easier, I think it's sufficient to only care about your mutex cleaning in encoded motor.

Reply:
this hasn't affected my testing, I think everything works in this state for both encoded and controlled, but I'll double check again.

Reply:
I'm using bucket rover where one motor has controls and the other doesn't so I can test both at the same time

Reply:
great make sure you do all the interrupting teest where:

- [ ] A GoFor interrupts a GoFor on the RC Card
- [ ] SetVelocity Interrupts MoveStraight on the RC Card
- [ ] motor.SetPower Interrupts GoFor on the RC Card
- [ ] base.SetPower Interrupts MoveStraight On the RC Card
- [ ] You can still shut down gracefully after each routine is cancelled, no threads remain after close.

Reply:
second one doesn't work on stable so I'll try and fix that bug

Reply:
Add a new ticket and fix it then.